modules/express/src/expressApp.ts

```diff
@@ -196,7 +202,7 @@ export function startup(config: Config, baseUri: string): () => void {
 /**
  * helper function to determine whether we should run the server over TLS or not
  */
-function isTLS(config: Config): boolean {
+function isTLS(config: Config): config is Config & { keyPath: string; crtPath: string } {
```

Comment:
I think this function should still return a boolean and not a config object.

Reply:
This might just be really advanced TypeScript stuff, would you mind to explain?

Reply:
Sure, this is called a "User Defined Type Guard": https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

This function still returns a boolean, but this is a way to tell the compiler "if this function returns true, then we can have some more guarantees about the parameter type". More info here: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
In this case, if `isTLS` returns true, we know the config must have the `keyPath` and `crtPath` properties, so we can make tell the compiler that these properties are now non-optional.